docs/ERC721.md

```diff
@@ -0,0 +1,686 @@
+
+### Supporting Interfaces
+
+In order to attain seamless layer one interoperability, this ERC721 implementation uses hardcoded interface IDs that have already been calculated on layer one. The Solidity implementation of ERC165 uses the `super` method to amend parent functions. Being that Cairo/Starknet currently does not support this feature, this implementation uses a mapping to store supported interfaces (similar to OpenZeppelin's [ERC165Storage](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/introspection/ERC165Storage.sol)). 
+
+## Extensibility
+
+This implementation of Cairo smart contract extensibility is set up to include all ERC721 storage and function logic within the base contract `ERC721_base.cairo`. Base contract methods with the prefix `ERC721_` must be imported to the frontend contract and inserted into an `external` method with the requisite name. This is already done in `ERC721.cairo`; however, additional functionality can be added. For instance, you could:
+- Implement a pausing mechanism
+- Add roles such as owner or minter
+- Modify the `transferFrom` function to perform actions [before](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol#L335)
+
+Just be sure that the exposed `external` methods invoke their imported function logic a la `approve` invokes `ERC721_approve`. As an example, see below.
+
+```python
+from contracts.token.ERC721_base import ERC721_approve
+
+@external
+func approve{
+        pedersen_ptr: HashBuiltin*, 
+        syscall_ptr: felt*, 
+        range_check_ptr
+    }(to: felt, token_id: Uint256):
+    ERC721_approve(to, token_id)
+    return()
+end
+
+```
+
+### Mintable
+
+The ERC721_Mintable preset offers a quick and easy setup for creating NFTs. This preset includes two imports: Ownable and ERC721_base. Further, this contract includes `mint` and `burn` methods to create and destroy NFTs. A common implementation of ERC721 contracts include setting a check on the `mint` and `burn` methods; where, only the owner can create and destroy tokens. In this preset, the owner is set in the constructor. 
+
+### Metadata
+
+The Metadata extension is quite similar to the Mintable preset; however, this preset includes a baseURI parameter in the constructor and omits the `burn` method. Setting the baseURI initializes the `tokenURI` method to return an array with both the baseURI and the token_id. Without this set, `tokenURI` returns an array with a zero value. 
+
+
+### Enumerable
+
+The Enumerable extension adds enumerability of all the token ids in the contract as well as all token ids owned by each account. This allows contracts to publish its full list of NFTs and make them discoverable. In regard to implementation, the extension exposes three new view methods as seen in the interface. Tokens are tracked via the `all_tokens_list` and `all_tokens_len`. No additional changes are required in the constructor for deployment.
```

Comment:
Reading about a given preset it's hard with the explanation here and the interfaces above. Do you think it's worth putting them in here instead, or just linking from here?